### PR TITLE
[task-3] : testing services implementation

### PR DESCRIPTION
**Pull request by zarutski**

### CODE REVIEW

File: song/src/test/java/com/epam/learn/song/component/SongServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.Map;
+
+@TestComponent
+@FeignClient(value = "song-service", url = "http://localhost:8093")
```
Comment (uladzimir-shelhunou) [inline song/src/test/java/com/epam/learn/song/component/SongServiceClient.java:15]: Why do you need a separate client for testing? You should use existing clients.

File: song/src/test/java/com/epam/learn/song/component/StepsDefinitions.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Autowired
+    SongServiceClient songServiceClient;
+
+    private ResponseEntity<Song> getResponse;
+
+    private ResponseEntity<Map<String, Integer>> postResponse;
+
+    @When("song's data saved")
+    public void song_s_data_saved(Song song) {
```
Comment (uladzimir-shelhunou) [inline song/src/test/java/com/epam/learn/song/component/StepsDefinitions.java:25]: Naming conversions are weird.

File: song/src/test/java/com/epam/learn/song/component/SongDataTableType.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+
+import java.util.Map;
+
+@Configurable
+public class SongDataTableType {
+
+    @Autowired
+    private ObjectMapper objectMapper;
```
Comment (uladzimir-shelhunou) [inline song/src/test/java/com/epam/learn/song/component/SongDataTableType.java:15]: Use constructor injection.

**Review by uladzimir-shelhunou — CHANGES_REQUESTED** [review]

File: song/src/test/java/com/epam/learn/song/component/SongDataTableType.java (hunk 3 above)
Comment (zarutski) [reply song/src/test/java/com/epam/learn/song/component/SongDataTableType.java]: Replaced

File: song/src/test/java/com/epam/learn/song/component/StepsDefinitions.java (hunk 2 above)
Comment (zarutski) [reply song/src/test/java/com/epam/learn/song/component/StepsDefinitions.java]: Fixed

File: song/src/test/java/com/epam/learn/song/component/SongServiceClient.java (hunk 1 above)
Comment (zarutski) [reply song/src/test/java/com/epam/learn/song/component/SongServiceClient.java]: Removed

**Review by uladzimir-shelhunou — APPROVED** [review]

**Pull request merged**